packages/python/javdb_integrations/email_notification.py

```
@@ -908,6 +908,80 @@ def _is_redownload_dedup_reason(reason):
+    Returns a dict with keys ``start_time``, ``end_time``, ``pending``,
+    ``purged``, ``failed``, ``skipped`` for the LAST such block, or
+    ``None`` when the log is missing / no completed run found.
+    """
+    path = log_path or DEDUP_LOG_FILE
+    if not path or not os.path.exists(path):
+        return None
+
+    try:
+        with open(path, 'r', encoding='utf-8', errors='replace') as f:
+            lines = f.readlines()
+    except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)